File: Binary Search/codeforces_379072H.cpp

// Problem link ===>> https://codeforces.com/group/isP4JMZTix/contest/379072/problem/H
// submission link ===>> https://codeforces.com/group/isP4JMZTix/contest/379072/submission/312669019

#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define endl '\n'

int isOk(int mid, int target, vector<int> &nums)
{
    if (nums[mid] <= target)
        return 0;
    else
        return 1;
}

int main()
{
    ios_base::sync_with_stdio(false);
    cout.tie(nullptr);
    cin.tie(nullptr);

    ll n, m;
    cin >> n >> m;

    vector<int> a(n), b(m);

    for (int i = 0; i < n; i++)
    {
        cin >> a[i];
    }
    for (int i = 0; i < m; i++)
    {
        cin >> b[i];
    }
    sort(a.begin(), a.end());

    for (auto num : b)
    {
        int l = 0, r = n;
        while (l < r)
        {
            int mid = (l + r) / 2;
            if (isOk(mid, num, a) == 0)
                l = mid + 1;
            else
                r = mid;
        }
        cout << l << " ";
    }
    return 0;
}

File: Binary Search/codeforces_379072H.py

# Problem link ===>> https://codeforces.com/group/isP4JMZTix/contest/379072/problem/H


import sys


def is_ok(mid, target, nums):
    return nums[mid] > target


def main():
    n, m = map(int, sys.stdin.readline().split())
    a = list(map(int, sys.stdin.readline().split()))
    b = list(map(int, sys.stdin.readline().split()))

    a.sort()

    result = []
    for num in b:
        l, r = 0, n
        while l < r:
            mid = (l + r) // 2
            if not is_ok(mid, num, a):
                l = mid + 1
            else:
                r = mid
        result.append(str(l))

    print(" ".join(result))


if __name__ == "__main__":
    main()
